portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -50,16 +61,18 @@ public void init() {
     quotes.add("\"From time to time I send Dwight faxes. From himself. From the future.\" - Jim Halpert");
     quotes.add("\"I disagree with.\" - Jim Halpert");
 
-    // Initialize the character's votes to 0
+    // Retrieve the character's votes from the datastore
     favoriteCharacterCount = new TreeMap<>();
-    try {
-      Scanner characterNamesScanner = new Scanner(new File("./character_names.txt"));
-      while(characterNamesScanner.hasNext()){
-        favoriteCharacterCount.put(characterNamesScanner.nextLine(), 0);
-      }
-    } catch (FileNotFoundException e) {
-      System.out.println("WARNING: character_names.txt couldn't be found");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(new Query("characterVotes"));
+    for(Entity entity : results.asIterable()){
+      String characterName = entity.getKey().getName();
+      int numVotes = (int)(long) entity.getProperty("numVotes");
```

Comment:
Are we casting this to a long and then an int? Not sure why we have two castings here? Can you please explain further please.

Reply:
Yeah this is a bit weird but I got runtime exceptions if I casted it straight to an int. The examples showed that they cast to a long so if I cast to a long and then an int, the exceptions go away. I can just make the variable a long if you'd prefer.